ouroboros-consensus/src/Ouroboros/Consensus/Protocol/Abstract.hs

```diff
@@ -166,6 +168,20 @@ class ( Show (ChainState    p)
   -- | We require that protocols support a @k@ security parameter
   protocolSecurityParam :: NodeConfig p -> SecurityParam
 
+  -- | The slot length, where known.
+  --
+  -- The duration of a slot is typically fixed for a certain protocol, but
+  -- might change when we switch from one protocol to another via a hard fork.
+  -- In that case, the 'LedgerView' can be used to detect if the hard fork has
+  -- happened.
+  --
+  -- Should guarantee that "later" calls (later on the blockchain) can extend,
+  -- but not modify, the information returned by earlier calls
```

Comment:
This comment is pretty unclear to me. The information returned is the slot length? How can this be "extend[ed], but not modif[ied]"?

Reply:
This is still a TODO, instead of returning `SlotLength`, something like `[(SlotNo, SlotLength)]` should be returned, indicating up to which slot the slot length is known. When it's decided when a hard fork will occur, this method should start including that information so we know in advance when the `SlotLength` will change. @edsko can chime in here.